pkgs/by-name/on/onedrivegui/package.nix

```diff
@@ -0,0 +1,66 @@
+{ lib
+, python3Packages
+, fetchFromGitHub
+, copyDesktopItems
+, onedrive
+, makeDesktopItem
+}:
+
+python3Packages.buildPythonApplication rec {
+  pname = "onedrivegui";
+  version = "1.0.3";
+
+  src = fetchFromGitHub {
+    owner = "bpozdena";
+    repo = "OneDriveGUI";
+    rev = "v${version}";
+    hash = "sha256-HutziAzhIDYP8upNPieL2GNrxPBHUCVs09FFxdSqeBs=";
+  };
+
+  nativeBuildInputs = [ copyDesktopItems ];
+
+  propagatedBuildInputs = [ onedrive ] ++
+  builtins.attrValues { inherit (python3Packages) pyside6 requests; };
+
+  doCheck = false; # No tests defined
+
+  desktopItems = [
+    (makeDesktopItem {
+      name = "OneDriveGUI";
+      exec = "onedrivegui";
+      desktopName = "OneDriveGUI";
+      comment = "OneDrive GUI Client";
+      type = "Application";
+      icon = "OneDriveGUI";
+      terminal = false;
+      categories = [ "Utility" ];
+    })
+  ];
+
+  preBuild = ''
+    cat > setup.py << EOF
+    from setuptools import setup
+    setup(
+      name='onedrivegui',
+      version='$version',
+      scripts=[
+        'src/OneDriveGUI.py',
+      ],
+    )
+    EOF
+  '';
+
+  postInstall = ''
+    ln -s $src/src/{resources,ui} $out/bin/
```

Comment:
This looks very strange - why are you pulling in the source?

Reply:
> This looks very strange - why are you pulling in the source?

The binary basically needs access to the subdirectories `resources` and `ui` which contain various assets. It was a choice between copying them into `$out/bin` or linking them from source. Is there a more canonical way to do this? Many thanks!

Reply:
Would this be better?

    cp -r $src/src/{resources,ui} $out/bin